### PR TITLE
Update MC Utils patch to expose ListTag like CompoundTag

### PR DESCRIPTION
**Pull request by Protonull**

### CODE REVIEW

**Comment by electronicboy** [comment]
1) uncommented
2) why?

**Comment by Protonull** [comment]
Sorry, fixing the comments now. CompoundTag has a similar change [here](https://github.com/PaperMC/Paper/blob/e295b608a2df686f5def6e2bd25dfd166e6b6d93/patches/server/0006-MC-Utils.patch#L2301-L2302) to make manipulation easier, so this patch allows for the same ease of manipulation.

File: patches/server/0006-MC-Utils.patch [hunk 1]
```diff
@@ -2314,6 +2314,23 @@ index 3d374000cd61d4a29dae21035c5ee9a93a1ff0f9..e59475b7bb3e000afece0033c5d3f112
...
+diff --git a/src/main/java/net/minecraft/nbt/ListTag.java b/src/main/java/net/minecraft/nbt/ListTag.java
+index 88bac72edf19c578902f49d20353989ed4d96f8f..014c67e18542264d877a7810366ca60f8059b163 100644
+--- a/src/main/java/net/minecraft/nbt/ListTag.java
++++ b/src/main/java/net/minecraft/nbt/ListTag.java
+@@ -50,10 +50,10 @@ public class ListTag extends CollectionTag<Tag> {
+         }
+     };
+     private static final ByteSet INLINE_ELEMENT_TYPES = new ByteOpenHashSet(Arrays.asList((byte) 1, (byte) 2, (byte) 3, (byte) 4, (byte) 5, (byte) 6)); // Paper - decompiler fix
+-    private final List<Tag> list;
++    public final List<Tag> list; // Paper - made public
```
Comment (Proximyst) [inline patches/server/0006-MC-Utils.patch:2326]: Do this with access transformers

**Review by Proximyst — CHANGES_REQUESTED** [review]
I don't personally see any reason to do this...

File: patches/server/0006-MC-Utils.patch [hunk 2]
```diff
@@ -2314,6 +2314,23 @@ index 3d374000cd61d4a29dae21035c5ee9a93a1ff0f9..e59475b7bb3e000afece0033c5d3f112
...
+     private byte type;
+ 
+-    ListTag(List<Tag> list, byte type) {
++    protected ListTag(List<Tag> list, byte type) { // Paper - made protected
```
Comment (Machine-Maker) [inline patches/server/0006-MC-Utils.patch:2330]: Use access transformers for this too

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

File: patches/server/0006-MC-Utils.patch (hunk 2 above)
Comment (Protonull) [reply patches/server/0006-MC-Utils.patch:2330]: What's an access transformer?

Comment (Machine-Maker) [reply patches/server/0006-MC-Utils.patch:2330]: Look at build-data/paper.at

**Comment by kennytv** [comment]
Closing this before you spend more time on this than necessary. This is silly considering you can already mutate the list in whatever way you want using existing methods. We sometimes do add "internal api", but that's only for things you are otherwise completely unable to achieve - which is definitely not the case here. The changes in the CompoundTag are just a byproduct of later patches and not meant for general use either.

**Pull request closed without merging**